pkg/controllers/scheduler/framework/runtime/framework_test.go

```diff
@@ -113,3 +203,112 @@ func TestRunFilterPlugins(t *testing.T) {
+		"score":          func(f framework.Handle) (framework.Plugin, error) { return scorePlugin, nil },
+		"select":         func(f framework.Handle) (framework.Plugin, error) { return selectPlugin, nil },
+		"replicas":       func(f framework.Handle) (framework.Plugin, error) { return replicasPlugin, nil },
+		"filterAndScore": func(f framework.Handle) (framework.Plugin, error) { return filterAndScorePlugin, nil },
+		"scoreAndSelect": func(f framework.Handle) (framework.Plugin, error) { return scoreAndSelectPlugin, nil },
+	}
+	tests := []struct {
+		name           string
+		enabledPlugins EnabledPlugins
+		expected       *frameworkImpl
+		shouldError    bool
+	}{
+		{
+			"enabled plugins are respected",
+			EnabledPlugins{
+				FilterPlugins:   []string{"filter"},
+				ScorePlugins:    []string{"score"},
+				SelectPlugins:   []string{"scoreAndSelect"},
+				ReplicasPlugins: []string{"replicas"},
+			},
+			&frameworkImpl{
+				filterPlugins:   []framework.FilterPlugin{filterPlugin},
+				scorePlugins:    []framework.ScorePlugin{scorePlugin},
+				selectPlugins:   []framework.SelectPlugin{scoreAndSelectPlugin},
+				replicasPlugins: []framework.ReplicasPlugin{replicasPlugin},
+			},
+			false,
+		},
+		{
+			"enabled plugins are respected 2",
+			EnabledPlugins{
+				FilterPlugins: []string{"filter", "filterAndScore"},
+				ScorePlugins:  []string{"score", "scoreAndSelect"},
+				SelectPlugins: []string{"scoreAndSelect", "select"},
+			},
+			&frameworkImpl{
+				filterPlugins: []framework.FilterPlugin{filterPlugin, filterAndScorePlugin},
+				scorePlugins:  []framework.ScorePlugin{scorePlugin, scoreAndSelectPlugin},
+				selectPlugins: []framework.SelectPlugin{scoreAndSelectPlugin, selectPlugin},
+			},
+			false,
+		},
+		{
+			"repeated plugins returns error",
+			EnabledPlugins{
+				FilterPlugins:   []string{"filter"},
+				ScorePlugins:    []string{"score", "score"},
+				SelectPlugins:   []string{"scoreAndSelect"},
+				ReplicasPlugins: []string{"replicas"},
+			},
+			nil,
+			true,
+		},
+		{
+			"incorrect type returns error",
+			EnabledPlugins{
+				FilterPlugins:   []string{"replicas"},
+				ScorePlugins:    []string{"score", "scoreAndSelect"},
+				SelectPlugins:   []string{"scoreAndSelect", "select"},
+				ReplicasPlugins: []string{"filter"},
+			},
+			nil,
+			true,
+		},
+		{
+			"plugins not found in registry returns error",
+			EnabledPlugins{
+				FilterPlugins:   []string{"filter"},
+				ScorePlugins:    []string{"score", "scoreAndSelect"},
+				SelectPlugins:   []string{"scoreAndSelect", "select", "notexists"},
+				ReplicasPlugins: []string{"replicas"},
+			},
+			nil,
+			true,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			fwk, err := NewFramework(registry, nil, &test.enabledPlugins)
+			if test.shouldError {
+				if err == nil {
+					t.Errorf("expected error when creating framework but got nil")
```

Comment:
```suggestion
					t.Fatalf("expected error when creating framework but got nil")
```
`Errorf` does not terminate the current test.

Reply:
fixed